src/main/java/de/hysky/skyblocker/skyblock/tabhud/config/preview/PreviewTab.java

```diff
@@ -0,0 +1,537 @@
+				}
+
+				@Override
+				protected void appendClickableNarrations(NarrationMessageBuilder builder) {
+				}
+			});
+		}
+
+		widgetOptions.addWidget(ButtonWidget.builder(Text.literal("Apply Everywhere"), button -> {
+					if (this.previewWidget.selectedWidget == null) return;
+					PositionRule toCopy = ScreenMaster.getScreenBuilder(getCurrentLocation()).getPositionRule(this.previewWidget.selectedWidget.getInternalID());
+					if (toCopy == null && !(this.previewWidget.selectedWidget instanceof TabHudWidget)) return;
+					for (Location value : Location.values()) {
+						if (value == getCurrentLocation() || value == Location.DUNGEON || value == Location.UNKNOWN) continue;
+						ScreenMaster.getScreenBuilder(value).setPositionRule(
+								this.previewWidget.selectedWidget.getInternalID(),
+								toCopy
+						);
+					}
+					button.setMessage(Text.literal("Applied!"));
+					Scheduler.INSTANCE.schedule(() -> button.setMessage(Text.literal("Apply Everywhere")), 15);
+				}).width(width).tooltip(Tooltip.of(Text.literal("Apply positioning to all locations. This cannot be restored!"))).build()
+		);
+	}
+
+	public Location getCurrentLocation() {
+		return mode == Mode.DUNGEON ? Location.DUNGEON : parent.getCurrentLocation();
+	}
+
+	public ScreenMaster.ScreenLayer getCurrentScreenLayer() {
+		return currentScreenLayer;
+	}
+
+	private static class WidgetOptionsScrollable extends ScrollableWidget {
+
+		private final List<ClickableWidget> widgets = new ArrayList<>();
+		private int height = 0;
+
+		public WidgetOptionsScrollable() {
+			super(0, 0, 0, 0, Text.literal("Widget Options Scrollable"));
+		}
+
+		@Override
+		protected int getContentsHeight() {
+			return height;
+		}
+
+		@Override
+		protected double getDeltaYPerScroll() {
+			return 6;
+		}
+
+		@Override
+		protected void renderContents(DrawContext context, int mouseX, int mouseY, float delta) {
+			height = 0;
+			for (ClickableWidget widget : widgets) {
+				widget.setX(getX() + 1);
+				widget.setY(getY() + 1 + height);
+
+				height += widget.getHeight() + 1;
+				if (!isVisible(widget.getY(), widget.getBottom())) continue;
+				widget.render(context, mouseX, mouseY + (int) getScrollY(), delta);
+
+			}
+		}
+
+		@Override
+		public boolean mouseClicked(double mouseX, double mouseY, int button) {
+			for (ClickableWidget widget : widgets) {
+				if (!isVisible(widget.getY(), widget.getBottom())) continue;
+				if (widget.mouseClicked(mouseX, mouseY + getScrollY(), button)) return true;
+			}
+			return super.mouseClicked(mouseX, mouseY, button);
+		}
+
+		@Override
+		protected void drawBox(DrawContext context) {
+		}
+
+		@Override
+		protected void appendClickableNarrations(NarrationMessageBuilder builder) {
+		}
+
+		void clearWidgets() {
+			widgets.clear();
+		}
+
+		void addWidget(ClickableWidget clickableWidget) {
+			widgets.add(clickableWidget);
+		}
+	}
+
+	private class AnchorSelectionWidget extends ClickableWidget {
```

Comment:
maybe this class can be made `static`?

Reply:
uses methods inside PreviewTab (do you want em do be made more visible using PreviewTab.this.whatever)